packages/client/components/ReviewRequestToJoinOrgModal.tsx

```diff
@@ -27,79 +30,74 @@ const query = graphql`
+// TODO: Create generic dialog components using tailwind https://github.com/ParabolInc/parabol/issues/8107
+const ReviewRequestToJoinOrgModal = (props: Props) => {
+  const {closePortal, queryRef} = props
+
   const data = usePreloadedQuery<ReviewRequestToJoinOrgModalQuery>(query, queryRef)
+
   const [selectedTeams, setSelectedTeams] = useState<string[]>([])
   const teams = data.viewer.teams
   const sortedTeams = teams
     .slice()
     .sort((a, b) => a.organization.name.localeCompare(b.organization.name))
 
-  return (
-    <div>
-      {sortedTeams.map((team) => {
-        const {id: teamId, name: teamName, organization, teamMembers} = team
-        const {name: orgName} = organization
-
-        // TODO: implement filter on API side
-        const disabled = !!teamMembers.find((member) => member.userId === requestCreatedBy)
-        const active = selectedTeams.includes(teamId) || disabled
-
-        const handleClick = () => {
-          if (disabled) return
-
-          if (active) {
-            setSelectedTeams((prevSelectedTeams) => prevSelectedTeams.filter((id) => id !== teamId))
-          } else {
-            setSelectedTeams((prevSelectedTeams) => [...prevSelectedTeams, teamId])
-          }
-        }
-
-        return (
-          <div className='mb-2 flex items-center text-base' key={teamId} onClick={handleClick}>
-            <Checkbox
-              active={active}
-              disabled={disabled}
-              className={active && !disabled ? `text-sky-500` : undefined}
-            />
-            <label className={`ml-2 cursor-pointer ${disabled ? 'opacity-[.38]' : ''}`}>
-              {teamName} | {orgName}
-            </label>
-          </div>
-        )
-      })}
-    </div>
-  )
-}
-
-interface Props {
-  closePortal: () => void
-  requestCreatedBy: string
-  email: string
-}
+  const {domainJoinRequest} = data.viewer
 
-// TODO: Create generic dialog components using tailwind https://github.com/ParabolInc/parabol/issues/8107
-const ReviewRequestToJoinOrgModal = (props: Props) => {
-  const {closePortal, email, requestCreatedBy} = props
+  if (!domainJoinRequest) {
+    return null
+  }
 
-  const queryRef = useQueryLoaderNow<ReviewRequestToJoinOrgModalQuery>(query)
+  const {createdBy, createdByEmail} = domainJoinRequest
 
   return (
     <DialogContainer>
       <DialogTitle>{'Add teammate'}</DialogTitle>
       <DialogContent>
         <div className={'mb-4 text-base'}>
-          Which teams would you like to add <strong>{email}</strong> to?
+          Which teams would you like to add <strong>{createdByEmail}</strong> to?
+        </div>
+
+        <div>
+          {sortedTeams.map((team) => {
+            const {id: teamId, name: teamName, organization, teamMembers} = team
+            const {name: orgName} = organization
+
+            // TODO: implement filter on API side
+            const disabled = !!teamMembers.find((member) => member.userId === createdBy)
```

Comment:
+1 I think something like `isUserCreatedTeam` would be easier to understand. Also, `some` instead of `find` would mean we can remove the `!!`